### PR TITLE
[8.x] Add default rules in conditional rules

### PR DESCRIPTION
**Pull request by bastien-phi**
PR https://github.com/laravel/framework/pull/38361 introduced `ConditionalRules` to conditionally add rules to the validation.

Imagine a validation like : 
```php
public function rules()
{
    return [
        'email' => [
            Rule::when($this->someComplexTest(), ['required']),
            Rule::when(!$this->someComplexTest(), ['nullable']),
        ]
    ];
}
```
           
This PR adds the ability to define the behaviour when the condition is false.

```php
public function rules()
{
    return [
        'email' => [
            Rule::when($this->someComplexTest(), ['required'], ['nullable']),
        ]
    ];
}
```

This behavior already exists in collections, conditionable, ...

### CODE REVIEW

File: src/Illuminate/Validation/Rule.php [hunk 1]
```diff
@@ -20,11 +20,12 @@ class Rule
      *
      * @param  callable|bool  $condition
      * @param  array|string  $rules
+     * @param  array|string  $default
      * @return \Illuminate\Validation\ConditionalRules
      */
-    public static function when($condition, $rules)
+    public static function when($condition, $rules, $default = [])
```
Comment (driesvints) [inline src/Illuminate/Validation/Rule.php:26]: This is a breaking change unfortunately.

Comment (bastien-phi) [reply src/Illuminate/Validation/Rule.php:26]: Few days too late !

I will try differently

Comment (bastien-phi) [reply src/Illuminate/Validation/Rule.php:26]: Just realized I did not change the right method...

@driesvints `Rule::when` signature method must remain the same ? What about `ConditionalRules::__construct` ?

Comment (driesvints) [reply src/Illuminate/Validation/Rule.php:26]: Ah well to be fair if it's a very recent released thing we could maybe do it this way. I'll let @taylorotwell decide.

Comment (bastien-phi) [reply src/Illuminate/Validation/Rule.php:26]: Should I revert https://github.com/laravel/framework/pull/38450/commits/1daa1b3b67b2b0cd38ad416357e86242b6404864 to to get initial changes only ?

Comment (driesvints) [reply src/Illuminate/Validation/Rule.php:26]: @bastien-phi feel free yeah. Sorry about that.

Comment (bastien-phi) [reply src/Illuminate/Validation/Rule.php:26]: No problems ! Thanks for the review ;-)

**Comment by bastien-phi** [comment]
Thanks !

**Pull request merged**